source/common/filesystem/win32/filesystem_impl.cc

```diff
@@ -124,25 +124,34 @@ std::string InstanceImplWin32::fileReadToEnd(const std::string& path) {
   if (illegalPath(path)) {
     throw EnvoyException(absl::StrCat("Invalid path: ", path));
   }
-
-  std::ios::sync_with_stdio(false);
-
-  // On Windows, we need to explicitly set the file mode as binary. Otherwise,
-  // 0x1a will be treated as EOF
-  std::ifstream file(path, std::ios::binary);
-  if (file.fail()) {
+  auto fd = CreateFileA(path.c_str(), GENERIC_READ,
```

Comment:
I guess it's worth a comment why we're using this method (and unit test if you can get one working? might not be worth it though since we have the integration tests that demonstrate why its needed)

Reply:
Added a comment here. Let me know what you think